Add refresh token function

ktp242 opened the pull request:
The PR adds the function of `refresh_access_token`. When it gets called, it will refresh the current access token and update it with the new access token. So when a request fails and gets the status code of 401, that indicates the access token is expired, we can then hit `/refresh_access_token` to get a new one.

To elaborate the logic a bit more, notice that when we do `OAUTH_CLIENT.auth_code.get_token` for getting the access token for the first time, it will return an instance of `ACCESS_TOKEN` class that is define by `OAuth2` gem. `ACCESS_TOKEN` has some methods we can utilize and one of them is `refresh`. So if we want to use `refresh`, we have to store the `ACCESS_TOKEN` instance somewhere. And here, we store it as a class variable.

ktp242 committed:
Added refresh action and added a route for it.

ktp242 committed:
Updated README.md

ktp242 committed:
Removed the route to refresh the access token.

ktp242 committed:
Removed unnecessary parameters from refresh_access_token

ktp242 committed:
Refined the refresh_access_token function.

nodanaonlyzuul commented on app/controllers/oauth_controller.rb:
```diff
@@ -40,21 +49,38 @@ def callback
         # TODO: we need an authorization to check if the user is on the white list of the scsbuster
         redirect_to session[:original_url]
       rescue
+        puts 'Failed to get access token.'
```
Can we take a pass through the app and replace `puts` statements with calls to `Rails.logger`?

ktp242 replied:
Sure!!! I didn't know the method is in Rails
So the syntax will be 
```
Rails.logger('Failed to get access token.')
```
Is this correct?

ktp242 committed:
Upddated logging methods.

nodanaonlyzuul committed:
Make OauthController#refresh_access_token a helper method.

Although it's a function defined in a Controller, it's not an action,
it's not a URL that a browser would ever hit

nodanaonlyzuul committed:
Replace string literal URLs with built in Rails "_path" helpers

ktp242 committed:
Updated the paths Rail's made variables.
